packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -1368,19 +636,19 @@ export class ProjectLifecycleManager {
     return this.metaState.needsCypressJsonMigration && Boolean(legacyConfigFileExist)
   }
 
-  private _pendingInitialize?: pDefer.DeferredPromise<FullConfig>
-
-  async initializeRunMode () {
+  async initializeRunMode (testingType: TestingType | null) {
     this._pendingInitialize = pDefer()
 
-    if (!this._currentTestingType) {
-      // e2e is assumed to be the default testing type if
-      // none is passed in run mode
-      this.setCurrentTestingType('e2e')
-    }
+    if (await this.waitForInitializeSuccess()) {
+      if (!this.metaState.hasValidConfigFile) {
+        return this.ctx.onError(getError('NO_DEFAULT_CONFIG_FILE_FOUND', this.projectRoot))
+      }
 
-    if (!this.metaState.hasValidConfigFile) {
-      return this.ctx.onError(getError('NO_DEFAULT_CONFIG_FILE_FOUND', this.projectRoot))
+      if (testingType) {
+        this.setCurrentTestingType(testingType)
+      } else {
+        this.setCurrentTestingType('e2e')
```

Comment:
Should we make `setCurrentTestingType()` async now and await it where it's used?